File: Client/src/app/_models/ticket.ts
import {Description} from "./description";
import {Employee} from "./employee";
import {Status} from "./status";
import {Priority} from "./priority";
import {Client} from "./client";

export interface Ticket {

  id: string;
  title: string;
  description: Description;
  createdAt: Date;
  updatedAt: Date;
  employee: Employee;
  client: Client;
  status: Status;
  priority: Priority;
  enabled: boolean;

}

File: Tickets/src/main/java/com/dissertation/Tickets/entities/Ticket.java
package com.dissertation.Tickets.entities;

import lombok.Data;
import org.springframework.data.mongodb.core.mapping.Document;

import javax.persistence.Id;
import java.time.LocalDateTime;

@Data
@Document("tickets")
public class Ticket {

    @Id
    private String id;
    private String title;
    private Description description;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private long employeeId;
    private long clientId;
    private Status status;
    private Priority priority;
    private boolean enabled;

}
